### PR TITLE
ADEN-4715 | add separate autoplay for resolved state

### PR DESCRIPTION
**Pull request by mleczey**
## Links:
- https://wikia-inc.atlassian.net/browse/ADEN-4715

### CODE REVIEW

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 1]
```diff
@@ -2,19 +2,21 @@
...
 	'ext.wikia.adEngine.slot.resolvedState',
+	'ext.wikia.adEngine.slotTweaker',
 	'ext.wikia.adEngine.video.uapVideo',
+	'ext.wikia.adEngine.video.videoSettings',
 	'wikia.document',
 	'wikia.log',
 	'wikia.throttle',
 	'wikia.window',
 	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
 ], function (uapContext,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js]: MAJOR Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js [hunk 2]
```diff
@@ -155,113 +117,138 @@ describe('ext.wikia.adEngine.slot.resolvedState', function () {
...
+		expect(rs.isResolvedState()).toEqual(true);
 	});
 
-	it('Should not update image if there is no background image (template without backgroundImage)', function () {
-		var params = {},
+	it('should not modify params if template does not support resolved state', function() {
+		spyOn(mocks.videoSettings, 'getParams');
+
+		var params = createIncorrectParams(),
 			rs = getModule();
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js]: MAJOR Remove the declaration of the unused 'rs' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "rs" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/AdEngine/js/video/player/porvata/porvata.js [hunk 3]
```diff
@@ -94,7 +95,7 @@ define('ext.wikia.adEngine.video.player.porvata', [
 					}
 				});
 
-				if (params.autoPlay) {
+				if (videoSettings.isAutoPlay()) {
 					muteFirstPlay(video, isFirstPlay);
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/player/porvata/porvata.js]: CRITICAL "muteFirstPlay" expects "1" arguments, but "2" were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)

File: extensions/wikia/AdEngine/js/video/uapVideo.js [hunk 4]
```diff
@@ -6,12 +6,13 @@ define('ext.wikia.adEngine.video.uapVideo', [
...
 	'ext.wikia.adEngine.video.player.ui.videoInterface',
 	'ext.wikia.adEngine.video.player.uiTemplate',
+	'ext.wikia.adEngine.video.videoSettings',
 	'wikia.document',
 	'wikia.log',
 	'wikia.throttle',
 	'wikia.window',
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
-], function (uapContext, adSlot, porvata, playwire, videoInterface, UITemplate, doc, log, throttle, win, mercuryListener) {
+], function (uapContext, adSlot, porvata, playwire, videoInterface, UITemplate, VideoSettings, doc, log, throttle, win, mercuryListener) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/uapVideo.js:15]: MAJOR Function has 12 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/slot/resolvedState.js [hunk 5]
```diff
@@ -63,61 +55,54 @@ define('ext.wikia.adEngine.slot.resolvedState', [
...
+	function setImage(videoSettings) {
+		var params = videoSettings.getParams();
+
 		if (templateSupportsResolvedState(params)) {
-			params = hasResolvedState(params) ? setResolvedState(params) : setDefaultState(params);
+			if (videoSettings.isResolvedState()) {
+				setResolvedState(params);
+			} else {
+				setDefaultState(params);
+				updateInformationAboutSeenDefaultStateAd(params);
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/slot/resolvedState.js]: CRITICAL "updateInformationAboutSeenDefaultStateAd" expects "0" arguments, but "1" were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 6]
```diff
@@ -2,16 +2,20 @@
...
 	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slot.resolvedState',
 	'ext.wikia.adEngine.slotTweaker',
 	'ext.wikia.adEngine.video.uapVideo',
+	'ext.wikia.adEngine.video.videoSettings',
 	'wikia.document',
 	'wikia.log',
 	'wikia.window',
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
 ], function (uapContext,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js]: MAJOR Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* CRITICAL 2 critical
* MAJOR 5 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaaDesktop.js#L14](https://github.com/Wikia/app/blob/8a94a154c46d52b39b5f0dfb598d79a63b8e64ef/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [bfaaMobile.js#L13](https://github.com/Wikia/app/blob/8a94a154c46d52b39b5f0dfb598d79a63b8e64ef/extensions/wikia/AdEngine/js/template/bfaaMobile.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaaDesktop.js#L14](https://github.com/Wikia/app/blob/6690db03bfbde3ca4afe6400bf08c8e3bd026162/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [bfaaMobile.js#L13](https://github.com/Wikia/app/blob/6690db03bfbde3ca4afe6400bf08c8e3bd026162/extensions/wikia/AdEngine/js/template/bfaaMobile.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaaDesktop.js#L14](https://github.com/Wikia/app/blob/77baf3097098b52d6cf6f1919ea79d4245f2abc9/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [bfaaMobile.js#L13](https://github.com/Wikia/app/blob/77baf3097098b52d6cf6f1919ea79d4245f2abc9/extensions/wikia/AdEngine/js/template/bfaaMobile.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/video/videoSettings.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/*global define*/
+define('ext.wikia.adEngine.video.videoSettings', [
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/video/videoSettings.js:2]: Should we call it videoSettingsFactory?

File: extensions/wikia/AdEngine/js/template/porvata.js [hunk 8]
```diff
@@ -1,8 +1,9 @@
 /*global define*/
 define('ext.wikia.adEngine.template.porvata', [
 	'ext.wikia.adEngine.video.player.porvata',
+	'ext.wikia.adEngine.video.videoSettings',
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
-], function (porvata, mercuryListener) {
+], function (porvata, VideoSettings, mercuryListener) {
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/template/porvata.js]: Can it start with lower case?

File: extensions/wikia/AdEngine/js/slot/resolvedState.js [hunk 9]
```diff
@@ -78,8 +63,6 @@ define('ext.wikia.adEngine.slot.resolvedState', [
 		if (params.image2 && params.image2.resolvedStateSrc) {
 			params.image2.element.src = params.image2.resolvedStateSrc;
 		}
-
-		return params;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolvedState.js:82]: :gem:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaaDesktop.js#L14](https://github.com/Wikia/app/blob/4e35ce7a7bf613ee349e450cf9e456f55a5f9319/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [bfaaMobile.js#L13](https://github.com/Wikia/app/blob/4e35ce7a7bf613ee349e450cf9e456f55a5f9319/extensions/wikia/AdEngine/js/template/bfaaMobile.js#L13): Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Pull request merged**